src/test/java/com/android/volley/CacheDispatcherTest.java

```diff
@@ -231,4 +235,39 @@ public void duplicateSoftExpiredCacheHit_successfulRequest() throws Exception {
         verify(mDelivery)
                 .postResponse(any(Request.class), any(Response.class), any(Runnable.class));
     }
+
+    @Test
+    public void processRequestNotifiesListener() throws Exception {
+        final AtomicBoolean listenerWasCalledOnStart = new AtomicBoolean(false);
+        final AtomicBoolean listenerWasCalledOnFinish = new AtomicBoolean(false);
+
+        RequestQueue.RequestEventListener listener =
```

Comment:
WDYT about a generic test helper class which just records the list of events that were received, and then asserting that the set of events (in order) received matches expectations?

At that point I don't even think you need an explicit listener object, but could just use a Mockito mock instead and verify (inOrder) the calls to onRequestEvent.

Reply:
Done